python_modules/libraries/dagster-dbt/dagster_dbt_tests/core/test_resources_v2.py

```diff
@@ -303,6 +306,42 @@ def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
     assert result.success
 
 
+@pytest.mark.skipif(
+    version.parse(dbt_version) < version.parse("1.7.9"),
+    reason="`dbt retry` with `--target-path` support is only available in `dbt-core>=1.7.9`",
+)
+def test_dbt_retry_execution(
+    test_jaffle_shop_manifest: Dict[str, Any], dbt: DbtCliResource
+) -> None:
+    @dbt_assets(manifest=test_jaffle_shop_manifest)
+    def my_dbt_assets(context: AssetExecutionContext, dbt: DbtCliResource):
+        test_jaffle_shop_path.joinpath(
+            os.environ["DAGSTER_DBT_PYTEST_XDIST_DUCKDB_DBFILE_PATH"]
+        ).unlink()
+
+        dbt_invocation = dbt.cli(["run"], context=context, raise_on_error=False)
+
+        assert not dbt_invocation.is_successful()
+        assert not list(dbt_invocation.stream())
+
+        yield from dbt.cli(["seed"], context=context).stream()
+        yield from replace(
```

Comment:
is this necessary when context=context is set on the line above?

Reply:
Yes, since the `manifest` and `dagster_dbt_translator` arguments to `DbtCliInvocation` are created in the `DbtCliResource.cli(...)` method, not the `DbtCliInvocation.run(...)` method.

We could change that behavior, but I was planning on circumventing the usage of `replace` altogether.